### ISSUE TITLE
Form labels not quite correct

### ISSUE THREAD

**Issue by Mitchell64** [fuel/core#758]
I've noticed a couple of issues with the form output regarding labels. 

1) When using check boxes, the "label" field output looks like the following:

```
    <tr>
        <td class=""><label for="">Colors</label></td>
        <td class="">
            <input  type="checkbox" id="colors_0" name="colors[0]" value="1">
            <label for="colors_0">Blue</label>
        </td>
    </tr>
```

My understanding is that the 1st label tag with for="" should not be there. I believe this was an issue quite a while ago and was fixed, but somewhere this error was reintroduced.

2) When the "required_mark" is used, I believe it is not displayed in the proper place. For example, here's some output:

```
    <th class="">
        <label for="title">Title</label>
        <abbr class="required" title="required">*</abbr>
    </th>
```

  I believe the required_mark should be inside the label tags like below:

```
    <th class="">
        <label for="title">Title
        <abbr class="required" title="required">*</abbr>
        </label>
    </th>
```

**Comment by darklow** [comment]
I totally agree with @Mitchell64 about the position of required mark. There is no way using current available template config to put required mark in the label field.

**Comment by jschreuder** [comment]
Closed, will be handled by issue #839.

**Comment by jschreuder** [comment]
This will be solved in 2.0 by introducing templating for form output.
